lib/saluki-components/src/destinations/datadog/metrics/request_builder.rs

```diff
@@ -397,29 +398,37 @@ fn encode_series_metric(metric: &Metric) -> proto::MetricSeries {
     let mut series = proto::MetricSeries::new();
     series.set_metric(metric.context().name().clone().into());
 
-    // Set our tags.
-    //
-    // This involves extracting some specific tags first that have to be set on dedicated fields (host, resources, etc)
-    // and then setting the rest as generic tags.
-    let mut tags = metric.context().tags().clone();
-
     let mut host_resource = Resource::new();
     host_resource.set_type("host".to_string().into());
     host_resource.set_name(metric.metadata().hostname().map(|h| h.into()).unwrap_or_default());
     series.mut_resources().push(host_resource);
 
-    if let Some(ir_tags) = tags.remove_tags("dd.internal.resource") {
-        for ir_tag in ir_tags {
-            if let Some((resource_type, resource_name)) = ir_tag.value().and_then(|s| s.split_once(':')) {
+    // Collect and handle all of our tags.
+    //
+    // We extract some specific tags and use them to add resource entries, such that they aren't sent as actual tags.
+    let mut tags = Vec::new();
+
+    let metric_tags = metric.context().tags();
```

Comment:
Woops, I forgot to update this area to use the `Tagged` impl I added to `Context`.